### PR TITLE
Add "Manhattan distance" as an additional distance metric.

### PR DESCRIPTION
**Pull request by sky-2002**
Currently, we can find cosine and l2-squared distance metrics. This PR attempts to add manhattan distance metric as well. This PR attempts to implement one of the distance metrics mentioned in [this issue](https://github.com/semi-technologies/weaviate/issues/1745).

### CODE REVIEW

**Comment by sky-2002** [comment]
I agree with the CLA

File: adapters/repos/db/vector/hnsw/config_test.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func Test_UserConfig(t *testing.T) {
 		input    interface{}
 		expected UserConfig
 	}
-
+	// these are the tests, I will have to add a test for manhattan distance as well
```
Comment (parkerduckworth) [inline adapters/repos/db/vector/hnsw/config_test.go:94]: This comment can be removed

File: adapters/repos/db/vector/hnsw/config_test.go [hunk 2]
```diff
@@ -160,8 +160,40 @@ func Test_UserConfig(t *testing.T) {
 			},
 		},
 
+		// ----trying to add test case for manhattan distance
```
Comment (parkerduckworth) [inline adapters/repos/db/vector/hnsw/config_test.go:163]: This comment can be removed

File: adapters/repos/db/vector/hnsw/config_test.go [hunk 3]
```diff
@@ -160,8 +160,40 @@ func Test_UserConfig(t *testing.T) {
...
+				FlatSearchCutoff:       16,
+				DynamicEFMin:           17,
+				DynamicEFMax:           18,
+				DynamicEFFactor:        19,
+				Skip:                   true,
+				Distance:               "manhattan",
+			},
+		},
+
+		// ----end of test case for manhattan distance------------------------------
```
Comment (parkerduckworth) [inline adapters/repos/db/vector/hnsw/config_test.go:194]: This comment can be removed

**Review by parkerduckworth — CHANGES_REQUESTED** [review]

File: adapters/repos/db/vector/hnsw/config_test.go (hunk 1 above)
Comment (sky-2002) [reply adapters/repos/db/vector/hnsw/config_test.go:94]: Yes, sure. Forgot to remove these, used them as reference while adding 😂.

File: adapters/repos/db/vector/hnsw/config_test.go (hunk 2 above)
Comment (sky-2002) [reply adapters/repos/db/vector/hnsw/config_test.go:163]: 👍

File: adapters/repos/db/vector/hnsw/config_test.go (hunk 3 above)
Comment (sky-2002) [reply adapters/repos/db/vector/hnsw/config_test.go:194]: Sure 👍

**Pull request merged**